### PR TITLE
Fix error handling

### PR DESCRIPTION
**Pull request by PeterBenc**
Motivation: 

- Lot of ledger related errors come to sentry daily, and all we can do is to advice users to try using webUsb or to our trouble shooting guide. 
- If a network error happens during the fee calculation it results in unexpected error since its not cached.

Changes: 

- catch errors when calculating fee
- add translations for general `Error` and modify the `captureBySentry` helper to send also the params into translations so if the for example the `Error` is present in the translations but doest have a translation matching its params it returns undefined thus resulting in the error being considered unexpected.
- add the ledger errors to `errorsWithHelp` so it automatically recommends user to read the troubleshooting guide, and if that doesn't help he can send us the error if he really wants to.

Gotcha: 

- We should probably refactor and type the translations and also sentry capturing logic a bit but this is mostly a hotfix so we don't have to respond manually to every user.
- I also update our troubleshooting guide and added a ledger section
- the `SyntaxError: Failed to execute 'postMessage' on 'Window'` is a issue for quite some time and I havent been able to find any fix or reasoning behind it happening, only that it has to do something with some build in chrome crypto extension. Anyway the only thing I  usually respond to the users is to checkout the trouble shooting guide anyway.

What user will see from now on:
 
image

### CODE REVIEW

File: app/frontend/translations.ts [hunk 1]
```diff
@@ -6,6 +6,9 @@ import {Lovelace, CryptoProviderFeature} from './types'
 
 const {ADALITE_MIN_DONATION_VALUE} = ADALITE_CONFIG
 
+const ledgerConnectError =
+  ' Please try connecting you device using the "Connect with WebUSB" functionality (the button underneath "Unlock with Ledger"). For more information please follow the Ledger part of our troubleshooting suggestions.'
```
Comment (refi93) [inline app/frontend/translations.ts:10]: probably redundant space at the beginning and typo - `you` -> `your`

Comment (PeterBenc) [reply app/frontend/translations.ts]: Fixed, thnx :+1:

File: app/frontend/translations.ts [hunk 2]
```diff
@@ -130,6 +136,15 @@ const translations = {
   PoolRegNoTtl: () =>
     'TTL parameter is missing in the transaction. It is explicitly required even for the Allegra era.',
   PoolRegTxParserError: ({message}) => `Parser error: ${message}`,
+
+  Error: ({message}) => {
+    const errors = {
+      'NotFoundError: The device was disconnected.': `${message}${ledgerConnectError}`,
+      'AbortError: The transfer was cancelled.': `${message}${ledgerConnectError}`,
+      "SyntaxError: Failed to execute 'postMessage' on 'Window': Invalid target origin 'chrome-extension://kmendfapggjehodndflmmgagdbamhnfd' in a call to 'postMessage'": `${message}`,
```
Comment (refi93) [inline app/frontend/translations.ts:144]: where does that error message come from? Not sure if that extension identifier is global

File: app/frontend/actions.ts [hunk 3]
```diff
@@ -603,17 +603,24 @@ export default ({setState, getState}: {setState: SetStateFn; getState: GetStateF
...
-    const plan: any = await wallet.getAccount(state.sourceAccountIndex).getTxPlan(args)
-    // FIXME: this cant be any
-    if (plan.error) {
-      stopLoadingAction(state, {})
-      resetDelegationWithoutHash(state)
-      setState({
-        calculatingDelegationFee: false,
-        calculatingFee: false,
-      })
+    try {
```
Comment (refi93) [inline app/frontend/actions.ts:606]: what's the purpose of this try/catch? what kind of error can happen here that it's required?

File: app/frontend/actions.ts [hunk 4]
```diff
@@ -603,17 +603,24 @@ export default ({setState, getState}: {setState: SetStateFn; getState: GetStateF
...
+      if (plan.error) {
+        stopLoadingAction(state, {})
+        resetDelegationWithoutHash(state)
+        setState({
+          calculatingDelegationFee: false,
+          calculatingFee: false,
+        })
+      }
+      return plan
+    } catch (e) {
```
Comment (refi93) [inline app/frontend/actions.ts:618]: I would restrict this to network errors if that's our concern to not silence other exceptions that may be e.g. related to some bug or other kind of malfuncion

File: app/frontend/translations.ts (hunk 2 above)
Comment (PeterBenc) [reply app/frontend/translations.ts:144]: I mentioned it briefly in the PR description, its pretty rare and regarding the extension identifier, I haven't checked it but I was researching it couple of times I always came to that one crypto-something extension in chrome that somehow causes this.

File: app/frontend/translations.ts [hunk 5]
```diff
@@ -6,6 +6,9 @@ import {Lovelace, CryptoProviderFeature} from './types'
 
 const {ADALITE_MIN_DONATION_VALUE} = ADALITE_CONFIG
 
+const ledgerConnectError =
```
Comment (refi93) [inline app/frontend/translations.ts:9]: ledgerTroubleshootingSuggestion is imho a more accurate name as it's not an error per se

File: app/frontend/translations.ts [hunk 6]
```diff
@@ -130,6 +136,15 @@ const translations = {
...
     'TTL parameter is missing in the transaction. It is explicitly required even for the Allegra era.',
   PoolRegTxParserError: ({message}) => `Parser error: ${message}`,
+
+  Error: ({message}) => {
+    const errors = {
+      'NotFoundError: The device was disconnected.': `${message}${ledgerConnectError}`,
+      'AbortError: The transfer was cancelled.': `${message}${ledgerConnectError}`,
+      "SyntaxError: Failed to execute 'postMessage' on 'Window': Invalid target origin 'chrome-extension://kmendfapggjehodndflmmgagdbamhnfd' in a call to 'postMessage'": `${message}`,
+    }
+    return errors[message]
```
Comment (refi93) [inline app/frontend/translations.ts:146]: this can get resolved to undefined if the message is lacking which is probably not desired, something like

```
return (message in errors) ? errors[message] ? message
```

Comment (PeterBenc) [reply app/frontend/translations.ts:146]: Actually, the undefined behavior is desired, we check whether an error has a translation and if not we display unexpectedErrorModal, so its crucial this returns undefined, otherwise all generic errors with the name `Error` would be "expected".

File: app/frontend/actions.ts (hunk 4 above)
Comment (PeterBenc) [reply app/frontend/actions.ts:618]: You are right, I didnt get to respond to the previous comment but our concern are the network errors indeed. All the other errors that might occur would still be displayed somewhere but you are right that they wouldn't be "unexpected". I ll add  
```
If (e.name !== 'NetworkError` || e.name !== 'ServerError') throw e
```

File: app/frontend/translations.ts (hunk 6 above)
Comment (refi93) [reply app/frontend/translations.ts:146]: I see - that's probably worth explicitly acknowledging with a comment so nobody is tempted to "fix" the function interface

File: app/frontend/translations.ts (hunk 2 above)
Comment (refi93) [reply app/frontend/translations.ts:144]: I see, it's apparently the CryptoToken extension allowing 2-step verification, likely via U2F: https://askubuntu.com/questions/844090/what-is-cryptotokenextension-in-chromium-extensions
From sentry it seems that the users that had that issue so far were using Ledger but of course, not necessarily. Maybe I would reword the troubleshooting suggestion to "If you are using Ledger Nano, ..." and put it here as well

File: app/frontend/actions.ts (hunk 4 above)
Comment (PeterBenc) [reply app/frontend/actions.ts:618]: I definitely plan to refactor this before the multi-asset so this whole tx plan creation works a bit better :D

File: app/frontend/translations.ts (hunk 2 above)
Comment (refi93) [reply app/frontend/translations.ts:144]: and please clarify with a comment where that arbitrary string comes from and why it's here

File: app/frontend/translations.ts (hunk 6 above)
Comment (PeterBenc) [reply app/frontend/translations.ts:146]: You are right, thanks for the suggestion, we should refactor this as well, but as mentioned in the description, this is rather a hotfix.

File: app/frontend/translations.ts (hunk 2 above)
Comment (PeterBenc) [reply app/frontend/translations.ts:144]: Thanks, I should have commented it. :+1:

File: app/frontend/translations.ts (hunk 6 above)
Comment (refi93) [reply app/frontend/translations.ts:146]: hotfixes without appropriate TODO comments have high chances of becoming permanent fixes :D

Comment (PeterBenc) [reply app/frontend/translations.ts:146]: True.. :D

**Comment by PeterBenc** [comment]
PS: I ll squash the commits before merging.

File: app/frontend/translations.ts [hunk 7]
```diff
@@ -6,6 +6,9 @@ import {Lovelace, CryptoProviderFeature} from './types'
 
 const {ADALITE_MIN_DONATION_VALUE} = ADALITE_CONFIG
 
+const ledgerTroubleshootingSuggestion =
+  'If you are using Ledger, please try connecting your device using the "Connect with WebUSB" functionality (the button underneath "Unlock with Ledger").For more information please follow section concerning Ledger in our troubleshooting suggestions.'
```
Comment (refi93) [inline app/frontend/translations.ts:10]:
```suggestion
  'If you are using Ledger, please try connecting your device using the "Connect with WebUSB" functionality (the button underneath "Unlock with Ledger"). For more information please check the section concerning Ledger in our troubleshooting suggestions.'
```

File: app/frontend/translations.ts [hunk 8]
```diff
@@ -130,6 +137,19 @@ const translations = {
...
+  Error: ({message}) => {
+    const errors = {
+      'NotFoundError: The device was disconnected.': `${message}${ledgerTroubleshootingSuggestion}`,
+      'AbortError: The transfer was cancelled.': `${message}${ledgerTroubleshootingSuggestion}`,
+      // an issue with CryptoToken extension allowing 2-step verification
+      // https://askubuntu.com/questions/844090/what-is-cryptotokenextension-in-chromium-extensions
+      "SyntaxError: Failed to execute 'postMessage' on 'Window': Invalid target origin 'chrome-extension://kmendfapggjehodndflmmgagdbamhnfd' in a call to 'postMessage'": `${message}${ledgerTroubleshootingSuggestion}`,
+    }
+    // we return undefined in case of unmached message on purpose since we
+    // want to threat such errors as unexpected
```
Comment (refi93) [inline app/frontend/translations.ts:150]:
```suggestion
    // want to treat such errors as unexpected
```

File: app/frontend/actions.ts [hunk 9]
```diff
@@ -603,17 +603,25 @@ export default ({setState, getState}: {setState: SetStateFn; getState: GetStateF
...
+      if (e.name !== 'NetworkError' || e.name !== 'ServerError') throw e
```
Comment (refi93) [inline app/frontend/actions.ts:619]: I would rather wrap the statement in curly braces so the return statement right below doesn't get confused as the logic triggered by the condition

```suggestion
      if (e.name !== 'NetworkError' || e.name !== 'ServerError') {
        throw e
      }

```

**Review by refi93 — APPROVED** [review]
last few nitpicks, otherwise LGTM
